microcosm_elasticsearch/tests/test_polymorphism.py

```diff
@@ -0,0 +1,164 @@
+"""
+Test polymorphic loading of subtypes.
+
+"""
+from math import pi
+
+from elasticsearch_dsl import Keyword
+from hamcrest import (
+    assert_that,
+    contains,
+    has_length,
+    instance_of,
+    is_,
+)
+from microcosm.api import binding, create_object_graph
+
+from microcosm_elasticsearch.models import Model
+from microcosm_elasticsearch.searching import SearchIndex
+from microcosm_elasticsearch.store import Store
+
+
+@binding("shape_index")
+def create_shape_index(graph):
+    return graph.elasticsearch_index_registry.register(version="v1")
+
+
+@binding("shape_search_index")
+class ShapeSearchIndex(SearchIndex):
+
+    def __init__(self, graph):
+        super().__init__(graph, graph.shape_index)
+
+
+class Shape(Model):
```

Comment:
It seems like `elasticsearch-dsl` respects the class name to define the doc type for a subclass, though I haven't found where it does that yet.

Previously we had defined our own, custom `__doctype_name__` extension, which I've stopped using here. But I think there are a few other ways to define the doc type and I haven't tracked them all down yet.

Reply:
I'd defined the default to be the class name (see [here](https://github.com/globality-corp/microcosm-elasticsearch/blob/0f09b3a9c679d7514aa7e1e179860d43d2b0ba93/microcosm_elasticsearch/models.py#L37))